File: week_challenge/typescript/solucionReto#3.ts
// solucion reto#3
// dificultad : media

// Es un numero prime ? 

//Enunciado: Escribe un programa que se encargue de comprobar si un número es o no primo.
// Hecho esto, imprime los números primos entre 1 y 100.

const isPrime = (number:number):void => {
    
    let index:number = 1;
    let result:number[] = [];

    for (index; index <= number; index++) {

        let isRemainderZero:boolean = number%index === 0;
        if (isRemainderZero ) result.push(number);
    }

    if (result.length === 2) console.log(number);
};

const oneToOneHundred = ():void => {

    let index:number = 1;

    while (index <= 100 ) {
        isPrime(index);
        index++;
    }
}

oneToOneHundred();

//big(0) time complexity = 0(n^^2)
//big(0) space complexity = 0(n^^2)

File: week_challenge/python/solucionReto#3.py
# solucion Reto#2
# Dificultad: DIFÍCIL

# LA SUCESIÓN DE FIBONACCI

# Enunciado: Escribe un programa que imprima los 50 primeros números de la sucesión de Fibonacci empezando en 0.
# La serie Fibonacci se compone por una sucesión de números en la que el siguiente siempre es la suma de los dos anteriores.
# 0, 1, 1, 2, 3, 5, 8, 13...


def isPrime(number):
    
    index = 1
    result = []

    while index <= number:
        isRemainderZero = number%index == 0
        if isRemainderZero :
             result.append(index)
        index+=1
    if len(result) == 2 :
        print(number)


def onetoHundred():

    index = 1
    while index <= 100:
        isPrime(index)
        index+=1
onetoHundred()

# big(0) time complexity = 0(n^^2)
# big(0) space complexity = 0(n^^2)